Adding instructions on removing LTSS packages

asettle opened the pull request:
Closes #715

asettle commented on xml/admin_ceph_upgrade.xml:
```diff
@@ -1109,6 +1109,28 @@ mds standby for name = mds.456-another-mds
+     before proceeding:
+    </para>
+<screen>&prompt.sminion;zypper rm -y sles-ltss-release sles-ltss-release-POOL</screen>
+   </important>
```
The only issue I have with this addition is we're saying "ensure these packages are installed but if they are, uninstall them" without a lot of context. Personally, I'd prefer some contextual understanding here? 
That, or perhaps we include it as part of the instruction set and add new steps:
1. Check if these packages are installed <command>
2. Optional: uninstall these packages <command>

smithfarm replied:
Before the starting the upgrade process, check whether the `sles-ltss-release` and/or `sles-ltss-release-POOL` packages are installed:

```
# rpm -q sles-ltss-release
# rpm -q sles-ltss-release-POOL
```

If either or both are installed, remove them:

```
# zypper rm -y PACKAGE
```

asettle replied:
I'll take that as an agreement to my second suggestion ;) I like this much better.

smithfarm replied:
(added two important caveats to the originally proposed text)

lproven commented on xml/admin_ceph_upgrade.xml:
```diff
@@ -1109,6 +1109,28 @@ mds standby for name = mds.456-another-mds
    </para>
+   <important>
+    <para>
+     Before starting the upgrade process, ensure the following packages are not installed:
```
```suggestion
     Before starting the upgrade process, ensure the following packages are <emphasis>not</emphasis> installed:
```

smithfarm commented:
@lproven @holgis and other reviewers: My understanding is that Alex is going to re-do this along the lines of https://github.com/SUSE/doc-ses/pull/842#discussion_r545183871

So maybe hold off on the reviews until she gets to that?

asettle commented on xml/admin_ceph_upgrade.xml:
```diff
@@ -1109,6 +1109,22 @@ mds standby for name = mds.456-another-mds
     <link xlink:href="https://documentation.suse.com/suse-distribution-migration-system/1.0/single-html/distribution-migration-system/"/>
     to find both general and detailed information about DMS.
    </para>
+    <para>
```
I'm also tempted to put this in its own procedure with a title such as "Before you Begin" - any thoughts? I do think it is clear enough as it is, but if anyone sees that as being a good way to call it out.

smithfarm replied:
I agree. There will definitely be multiple things to check before starting the upgrade process. For example, we will need to emphasize (as always), that the upgrade must not be started until all the SES6 nodes are fully patched.

asettle replied:
Good point, I don't believe we do a decent job of highlighting that. I'll add the new section :)

smithfarm replied:
(If we forgot that one, we'd be in deep trouble, because then customers on unpatched SES6 would commence the upgrade process and it would blow up, and then they'd point to the documentation and say: "nothing here says we need to patch first!")

asettle replied:
Story of my life?

asettle commented:
![Screenshot from 2020-12-18 11-38-32](https://user-images.githubusercontent.com/1987994/102610727-938cb680-4125-11eb-8553-bbb48bef99a9.png)
This is how I've updated it. Added the before you begin as a sub-section. I think this looks better. I tried to play around with the callouts to see which looked better, this was my personal preference. Doing it as a procedure looked... odd.

smithfarm commented:
Whoa, I didn't realize that this is for the SES5 -> 6 upgrade. For that, we already have a section that says the nodes must be fully patched before starting.

I also didn't realize that there's a _separate_, 4th-level section for upgrading via the Distribution Migration System, and that that's what we're editing here.

It seems like having two separate sections with upgrade procedures is not really a good idea? Could the "upgrading via the Distribution Migration System" bits be integrated into the main upgrade procedure?

asettle commented:
@smithfarm yeah absolutely, admittedly why I've found this all a bit odd to begin with. But that's the beauty of reading this all from a XML POV.

Also, we should note we already also have this: https://documentation.suse.com/ses/6/single-html/ses-deployment/#upgrade-prepare-repos

asettle commented:
> Whoa, I didn't realize that this is for the SES5 -> 6 upgrade. For that, we already have a section that says the nodes must be fully patched before starting.

Which is why I've just cross-referenced the section, and not rewritten it.

> 
> I also didn't realize that there's a _separate_, 4th-level section for upgrading via the Distribution Migration System, and that that's what we're editing here.

Yep. 

> 
> It seems like having two separate sections with upgrade procedures is not really a good idea? Could the "upgrading via the Distribution Migration System" bits be integrated into the main upgrade procedure?

I think we're having a bit of an issue with the upgrade process in general being unclear, perhaps?

smithfarm commented:
@asettle Presenting various parts of the upgrade procedure in multiple sections sounds chaotic. Would it be easier for readers to understand and follow if we had a single chapter for the SES5 -> 6 upgrade procedure and organized it (i.e. the whole chapter) as a series of steps?

Moreover, I believe this bit about checking for the LTSS and getting rid of it if it's there applies to everyone doing the upgrade, not just folks doing it via some "Distribution Migration System".

asettle commented:
> @asettle Presenting various parts of the upgrade procedure in multiple sections sounds chaotic. Would it be easier for readers to understand and follow if we had a single chapter for the SES5 -> 6 upgrade procedure and organized it (i.e. the whole chapter) as a series of steps?

Definitely. But I'm sorry, it's not in scope to spend hours changing the way this upgrade procedure is displayed right now. 

> 
> Moreover, I believe this bit about checking for the LTSS and getting rid of it if it's there applies to everyone doing the upgrade, not just folks doing it via some "Distribution Migration System".

Quite. But should we not be combining similar/relevant information? Especially the way it is currently set out.

smithfarm commented:
I guess the main point I'm trying to make here is that it seems wrong to me to hide away the instruction to check for the LTSS and remove it if it's there in a subsection entitled "Node Upgrade Using the Distribution Migration System", as if this only applied to folks using some "Distribution Migration System". This instruction applies to everyone doing the upgrade.

timscotland commented:
> I guess the main point I'm trying to make here is that it seems wrong to me to hide away the instruction to check for the LTSS and remove it if it's there in a subsection entitled "Node Upgrade Using the Distribution Migration System", as if this only applied to folks using some "Distribution Migration System". This instruction applies to everyone doing the upgrade.

While I 100% agree that the documentation layout could be improved to make it more simple to follow, the removal of LTSS is only required in the DMS method (when DMS runs it checks the repositories and hence fails)   In the manual method you boot from ISO and that makes no reference to the existing repos (hence does not cause an issue)

smithfarm commented:
In that case, I suggest to:

1. structure the upgrade chapter as a series of steps
2. at the beginning of the chapter, explain that there are two methods of upgrading: DMS and ISO, what they are, what the difference between them is
3. label each step in the procedure so it's clear that the step applies to the "ISO method", "DMS method", or "both methods"

asettle commented:
@timscotland @smithfarm Really appreciative of the feedback you are providing. At this time, I am unable to provide suitable resources to the effort of restructuring this chapter. I propose that we include this note ASAP, closing the initial issue and covering any future customer cases, and open a new issue with Nathan's (much needed) suggested changes.

Is this a suitable compromise at this time? :)

smithfarm commented:
> Is this a suitable compromise at this time? :)

I'm very happy with my comment being spun off into issue #843 - thank you!

asettle commented:
Thanks @smithfarm ! :smile:  okay, with that in mind. Please proceed with reviews on this PR as it is, and I will look into an appropriate timeline for fixing these docs.
